python/paddle/distributed/fleet/base/fleet_base.py

```diff
@@ -159,6 +160,8 @@ def init(self, role_maker=None, is_collective=False):
                     "`role_maker` should be subclass of `RoleMakerBase`, but got {}".
                     format(type(role_maker)))
         self.strategy_compiler = StrategyCompiler()
+        if paddle.fluid.framework.in_dygraph_mode():
```

Comment:
What if a user has already called paddle.distributed.init_parallel_env() outside this function?

Reply:
It will cause errors.  If-check is add for this. Thanks.